packages/backend/src/accounting/psql/ledger-transfer/index.test.ts

```diff
@@ -49,6 +69,322 @@ describe('Ledger Transfer', (): void => {
+    let baseTransfer: CreateTransferArgs
+    let totalAssetSettlementBalance: bigint
+
+    const accountStartingBalance = 100n
+
+    beforeEach(async (): Promise<void> => {
+      const accounts = [account, peerAccount]
+
+      baseTransfer = {
+        transferRef: uuid(),
+        creditAccount: account,
+        debitAccount: peerAccount,
+        ledger: account.ledger,
+        amount: 10n
+      }
+
+      // Fund accounts
+      await Promise.all(
+        accounts.map((account) =>
+          createLedgerTransfer(
+            {
+              creditAccountId: account.id,
+              debitAccountId: settlementAccount.id,
+              amount: accountStartingBalance,
+              ledger: account.ledger
+            },
+            knex
+          )
+        )
+      )
+
+      totalAssetSettlementBalance =
+        BigInt(accounts.length) * accountStartingBalance
+    })
+
+    test.each`
+      timeoutMs    | expectedState                  | description
+      ${undefined} | ${LedgerTransferState.POSTED}  | ${'without timeout'}
+      ${1000n}     | ${LedgerTransferState.PENDING} | ${'with defined timeout'}
+    `(
+      'properly creates $expectedState transfer $description',
+      async ({
+        timeoutMs,
+        expectedState
+      }: {
+        timeoutMs: bigint
+        expectedState: LedgerTransferState
+      }): Promise<void> => {
+        const transfer: CreateTransferArgs = {
+          ...baseTransfer,
+          timeoutMs,
+          type: LedgerTransferType.DEPOSIT
+        }
+
+        const now = new Date()
+
+        jest.useFakeTimers({ now })
+
+        await expect(
+          createTransfers(serviceDeps, [transfer], knex)
+        ).resolves.toEqual({
+          results: [
+            {
+              id: expect.any(String),
+              transferRef: transfer.transferRef,
+              creditAccountId: transfer.creditAccount.id,
+              debitAccountId: transfer.debitAccount.id,
+              ledger: transfer.ledger,
+              state: expectedState,
+              amount: transfer.amount,
+              type: transfer.type,
+              expiresAt: timeoutMs
+                ? new Date(now.getTime() + Number(timeoutMs))
+                : null,
+              createdAt: expect.any(Date),
+              updatedAt: expect.any(Date)
+            }
+          ],
+          errors: []
+        })
+      }
+    )
+
+    test('returns error if duplicate transfer (on unique transferRef constraint)', async (): Promise<void> => {
+      await expect(
+        createTransfers(serviceDeps, [baseTransfer, baseTransfer], knex)
+      ).resolves.toEqual({
+        results: [],
+        errors: [{ index: 0, error: TransferError.TransferExists }]
+      })
+    })
+
+    test('throws if unhandled error during insert', async (): Promise<void> => {
+      const transfer: CreateTransferArgs = {
+        ...baseTransfer
+      }
+
+      transfer.creditAccount.id = ''
```

Comment:
Should we be handling invalid account ids?

Reply:
This shouldn't happen since we fetch them directly from the DB first, but this would fail anyway on the insert